### PR TITLE
Add KV store providers (dynamic configuration only)

### LINKED ISSUE #5169: Add a provider for Key Value (KV) Store in Traefik v2
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

Traefik v1 supports dynamic config generation for and from a Key Value Store, i.e. it has a kv provider.

Now that the main changes for v2 are out, we can re-add previously supported providers, such as the one for Key Value Stores.

The existing providers, can be used for inspiration.

### PR DESCRIPTION
**Pull request by ldez**
### What does this PR do?

Allows to use a KV store as source of dynamic configuration.

The following KV stores are supported:
- Consul
- etcd
- Redis
- ZooKeeper

### Motivation

Fixes #5169,  #4369

### More

- [x] Added/updated tests
- [x] Added/updated documentation

### Additional Notes

Co-authored-by: Jean-Baptiste Doumenjou <jb.doumenjou@gmail.com>

### CODE REVIEW

File: pkg/provider/aggregator/aggregator.go [hunk 1]
```diff
@@ -53,6 +53,26 @@ func NewProviderAggregator(conf static.Providers) ProviderAggregator {
 		p.quietAddProvider(conf.ConsulCatalog)
 	}
 
+	if conf.Consul != nil {
+		p.quietAddProvider(conf.Consul)
+	}
+
+	if conf.Consul != nil {
```
Comment (dtomcej) [inline pkg/provider/aggregator/aggregator.go]: Duplicate?

File: integration/resources/compose/redis.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+redis:
+  image: redis:5.0
+  ports:
+    - "6379:6379"
```
Comment (dtomcej) [inline integration/resources/compose/redis.yml]: newline

File: pkg/config/static/static_config.go [hunk 3]
```diff
@@ -156,15 +160,21 @@ func (t *Tracing) SetDefaults() {
...
+	KubernetesIngress *ingress.Provider       `description:"Enable Kubernetes backend with default settings." json:"kubernetesIngress,omitempty" toml:"kubernetesIngress,omitempty" yaml:"kubernetesIngress,omitempty" export:"true" label:"allowEmpty"`
+	KubernetesCRD     *crd.Provider           `description:"Enable Kubernetes backend with default settings." json:"kubernetesCRD,omitempty" toml:"kubernetesCRD,omitempty" yaml:"kubernetesCRD,omitempty" export:"true" label:"allowEmpty"`
+	Rest              *rest.Provider          `description:"Enable Rest backend with default settings." json:"rest,omitempty" toml:"rest,omitempty" yaml:"rest,omitempty" export:"true" label:"allowEmpty"`
+	Rancher           *rancher.Provider       `description:"Enable Rancher backend with default settings." json:"rancher,omitempty" toml:"rancher,omitempty" yaml:"rancher,omitempty" export:"true" label:"allowEmpty"`
+	ConsulCatalog     *consulcatalog.Provider `description:"Enable ConsulCatalog backend with default settings." json:"consulCatalog,omitempty" toml:"consulCatalog,omitempty" yaml:"consulCatalog,omitempty"`
+
```
Comment (dtomcej) [inline pkg/config/static/static_config.go]: remove empty line?

Comment (ldez) [reply pkg/config/static/static_config.go]: I added the empty line to separate KV and other providers.

**Pull request merged**